engine/pkg/rpcutil/middleware.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpcutil
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	perrors "github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"github.com/pingcap/tiflow/pkg/errors"
+	"go.uber.org/zap"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ToGRPCError converts an error to a gRPC error.
+func ToGRPCError(errIn error) error {
```

Comment:
Please review this file.

Reply:
Should `ToGRPCError` and `FromGRPCError` be moved to `tiflow/pkg/errors` for better reuse by other modules?

Reply:
It involves engine API details. I think limiting it in engine's rpcutil is better. If someone want to adopt same API format as engine, he can extract these two functions outside.